src/Symfony/Component/Form/Tests/Extension/Core/Type/MoneyTypeTest.php

```diff
@@ -123,4 +124,28 @@ public function testHtml5EnablesSpecificFormatting()
         $this->assertSame('12345.60', $form->createView()->vars['value']);
         $this->assertSame('number', $form->createView()->vars['type']);
     }
+
+    public function testDefaultModelType()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, ['divisor' => 100]);
+        $form->submit('12345.67');
+
+        $this->assertSame(1234567.0, $form->getData());
+    }
+
+    public function testIntegerModelType()
+    {
+        $form = $this->factory->create(static::TESTED_TYPE, null, ['divisor' => 100, 'model_type' => 'integer']);
+        $form->submit('12345.67');
+
+        $this->assertSame(1234567, $form->getData());
+    }
+
+    public function testIntegerModelTypeExpectsDivisorNotEqualToOne()
+    {
+        $this->expectException(LogicException::class);
```

Comment:
should be moved before the submit()

Reply:
As you can see in one of the commits, that doesn't work. The exception is raised before. So maybe I could remove the submit from the tests

Reply:
that'd make sense yes since the line is never executed